src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace SourceGenerators.Tests
+{
+    internal static class RoslynTestUtils
```

Comment:
Was this file copied from somewhere? If so, where from? Should we put a comment saying where it was copied from?

Or even better, can we put it in a NuGet package and only have 1 copy of it?

Reply:
> Was this file copied from somewhere? If so, where from? Should we put a comment saying where it was copied from?
> 
> Or even better, can we put it in a NuGet package and only have 1 copy of it?

I created the file for this project, it was not copied from anywhere. In my original sources for LoggingGenerator, it does exist twice though, once in the analyzer tests and once in the generator tests. 

I think it worked out reasonably well, it'd be useful to make this a general consumable thing I guess.

Reply:
Ok, that answers my question. I've seen something like this in the past for analyzers - so I wasn't sure if this was copied from somewhere else.

For example, it isn't this: https://github.com/dotnet/roslyn-analyzers/tree/main/src/Test.Utilities

Reply:
It was tailor made at first for logging generator. 

Commit history: https://github.com/dotnet/runtime/commits/e8db38a65dd4f50c11305cb285020e01b2a21186/src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/RoslynTestUtils.cs

I placed it under common folder in case any future source generators want to reuse its logic.

cc: @geeknoid